portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,21 +35,28 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private int numComments = 10;
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      //addMessages();
       Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
       PreparedQuery results = datastore.prepare(query);
 
       ArrayList<Comment> comments = new ArrayList<>();
+      
+      int displayNum = numComments;
       for (Entity entity : results.asIterable()) {
           String name = (String) entity.getProperty("name");
           String content = (String) entity.getProperty("content");
           Date timestamp = (Date) entity.getProperty("timestamp");
 
           Comment comment = new Comment(name, content, timestamp);
           comments.add(comment);
+
+          displayNum--;
+          if (displayNum == 0)
+            break;
       }
```

Comment:
I put numComments in getPost because selecting the number of comments shown is apart of the same form as posting a comment, so they both get submitted together. And yeah I need to add pagination but I don't know hwo to do that yet

Reply:
Also, could you have just check `if (comments.size()>=numComments) break;`

Reply:
you right. fixed!